src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -2981,6 +2981,13 @@ String getApiTailUrl(String tail) {
         return "/repos/" + getOwnerName() + "/" + name + tail;
     }
 
+    String getApiHeadUrl(String head) {
+        if (head.length() > 0 && !head.endsWith("/")) {
+            head = '/' + head;
+        }
+        return "/user/" + head + getOwnerName() + "/" + name;
+    }
+
```

Comment:
I don't think this is needed here.  Each of the methods below can hard code `"/user/starred/" + full_name`.

Reply:
Hey @bitwiseman, I will work on this issue now. Bit lost in creating the test data for test cases, but i will soon figure it out.

Reply:
Fixed.